src/sidebar/reducers/links.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Reducer for storing a "links" object in the Redux state store.
+ *
+ * The links object is initially null, and can only be updated by completely
+ * replacing it with a new links object.
+ *
+ * Used by serviceUrl.
+ */
+
+'use strict';
+
+/** Return the initial links. */
+function init() { return {links: null}; }
+
+/** Return updated links based on the given current state and action object. */
+function update(state, action) { return {links: action.newLinks}; }
```

Comment:
I've renamed the functions `updateLinks` and `updateLinksAction`:

```diff
commit d63373c6c79b1895d11a0df99198fafc02db70e0 (HEAD -> use-new-links-api)
Author: Sean Hammond <git@snhmnd.fastmail.com>
Date:   Tue Apr 18 13:54:12 2017 +0100

    Rename funcs

diff --git a/src/sidebar/reducers/links.js b/src/sidebar/reducers/links.js
index f8993638..d1cf17f8 100644
--- a/src/sidebar/reducers/links.js
+++ b/src/sidebar/reducers/links.js
@@ -13,15 +13,15 @@
 function init() { return {links: null}; }
 
 /** Return updated links based on the given current state and action object. */
-function update(state, action) { return {links: action.newLinks}; }
+function updateLinks(state, action) { return {links: action.newLinks}; }
 
 /** Return an action object for updating the links to the given newLinks. */
-function action(newLinks) {
+function updateLinksAction(newLinks) {
   return { type: 'UPDATE_LINKS', newLinks: newLinks };
 }
 
 module.exports = {
   init:    init,
-  update:  { UPDATE_LINKS: update },
-  actions: { updateLinks: action },
+  update:  { UPDATE_LINKS: updateLinks },
+  actions: { updateLinks: updateLinksAction },
 };
```

But I'm not sure whether adding an `update` variable that maps action names to function expressions is worthwhile:

The code **before**:

```javascript
/** Return the initial links. */
function init() { return {links: null}; }

/** Return updated links based on the given current state and action object. */
function updateLinks(state, action) { return {links: action.newLinks}; }

/** Return an action object for updating the links to the given newLinks. */
function updateLinksAction(newLinks) {
  return { type: 'UPDATE_LINKS', newLinks: newLinks };
}

module.exports = {
  init:    init,
  update:  { UPDATE_LINKS: updateLinks },
  actions: { updateLinks: updateLinksAction },
};
```

The code **after**:

```javascript
/** Return the initial links. */
function init() { return {links: null}; }


/** Return an action object for updating the links to the given newLinks. */
function updateLinksAction(newLinks) {
  return { type: 'UPDATE_LINKS', newLinks: newLinks };
}

var update = {
  /** Return updated links based on the given current state and action object. */
  UPDATE_LINKS: function (state, action) {
    return {links: action.newLinks};
  },
};

module.exports = {
  init:    init,
  update:  update,
  actions: { updateLinks: updateLinksAction },
};

```

This may be more consistent with other reducer modules but I think it makes the code longer and harder to read, and also makes the module less consistent with itself (`module.exports.actions` is an inline literal object, but the `module.exports.update` object is a variable defined above, the update function is an inline function expression but the action and init functions are named, top-level functions).

I'd vote for keeping `module.exports.update` as a simple inline literal object?